### PR TITLE
Per app indexes

### PR DESCRIPTION
**Pull request by anishp55**
this enables applications to send to a specific index, without having to modify the props.conf or tranforms.conf.  a SPLUNK_INDEX env var is set for the application which will then override the default SPLUNK_INDEX of the nozzle itself.

### CODE REVIEW

**Comment by aegershman** [comment]
If this works this seems like it would be incredibly convenient.

Any reason this hasn't been commented on in so long?

**Comment by dtregonning** [comment]
@aegershman agreed. No excuse, we haven't done the best job keeping up to date with community PRs and issue tracking.
We have started kicking off a refresh to the Nozzle this week, so we will work through whats outstanding here and get it released ASAP.

**Comment by aegershman** [comment]
I totally get it, there are priorities. And I apologize, I don't mean to be pushy or naggy. I'm just trying to gauge where things are at so I can make better decisions to enable my developers. If some of these issues/PRs will get merged, awesome; if not, that's understandable, it's just nice to know one way or another so I have answers for people. I wouldn't be doing my job if I didn't ~annoy~ comment on these PRs or open issues with use-case feedback/questions. Plus the worst that can happen is someone says "shut up go away" 😉 

I appreciate what you all do & contribute & thank you for having this be OSS.

**Comment by anishp55** [comment]
hey guys, 
    i would love for this to be merged in.  we use it at my org right now, could probably use some more robust testing thought.


thanks

**Comment by luckyj5** [comment]
Hi @anishp55 Hope you doing well. Pardon the delay, we just updated PCF Tile with some minor changes to support PAS 2.4. Will now be looking into outstanding issues and PRs and provide an update as we go.  Thank you!

**Comment by grimmpp** [comment]
I'm also very interested in this functionality. We run cloud foundry with multiple tenants and I would like to be able to separate the data for those tenants.

**Comment by anishp55** [comment]
@luckyj5 sounds good!

**Comment by luckyj5** [comment]
Hi @anishp55, can you please update this PR to be in synch with develop (latest)? We are in the review process and synch with develop would help to test and validate against our CI. Please let me know if any questions. 
Thank you!

**Comment by anishp55** [comment]
Sure!

**Comment by anishp55** [comment]
so i've updated the app and verify it builds, but i am unable to test it against my 2.5 foundation
when make deploy-nozzle is run i get the following error (the same happens if i try to push it by hand)

`   events/events.go:9:2: cannot find package "github.com/Sirupsen/logrus" in any of:
   	/tmp/gobuildpack.gopath170422783/.go/src/github.com/cloudfoundry-community/splunk-firehose-nozzle/vendor/github.com/Sirupsen/logrus (vendor tree)
   	/tmp/contents638472504/deps/0/go1.10.8/go/src/github.com/Sirupsen/logrus (from $GOROOT)
   	/tmp/gobuildpack.gopath170422783/.go/src/github.com/Sirupsen/logrus (from $GOPATH)
          **ERROR** Unable to compile application: exit status 1
   Failed to compile droplet: Failed to run finalize script: exit status 12`

**Comment by anishp55** [comment]
i dont have a public cf deployment to push this too, but i have verified it internally, also removing the build steps from your deploy part will make the testing slightly faster.

**Comment by luckyj5** [comment]
Thanks @anishp55. Will look at the updated changes and let you know if anything.

File: glide.lock [hunk 1]
```diff
@@ -55,7 +55,7 @@ imports:
   - jwriter
 - name: github.com/pkg/errors
   version: 645ef00459ed84a119197bfb8d8205042c6df63d
-- name: github.com/sirupsen/logrus
+- name: github.com/Sirupsen/logrus
```
Comment (luckyj5) [inline glide.lock]: Please change it back from `Sirupsen` to `sirupsen`. This was changed to lowercase on purpose, please refer to https://github.com/cloudfoundry-community/splunk-firehose-nozzle/pull/151 and https://github.com/sirupsen/logrus/pull/546 for details.

File: glide.yaml [hunk 2]
```diff
@@ -13,3 +13,4 @@ import:
 - package: github.com/kelseyhightower/envconfig
 - package: github.com/pkg/errors
   version: ~0.8.0
+- name: github.com/Sirupsen/logrus
```
Comment (luckyj5) [inline glide.yaml]: Please remove this - not needed.

**Comment by luckyj5** [comment]
Hi @anishp55, I have requested few minor changes, but rest looks good. Thank you!

File: glide.yaml (hunk 2 above)
Comment (anishp55) [reply glide.yaml]: i found this was needed to build/deploy on a 2.5 foundation using the latest provided buildpack.

Comment (luckyj5) [reply glide.yaml]: Okay. I'll check with Pivotal as well on this and make adjustments later if needed. Thanks!

Comment (anishp55) [reply glide.yaml]: i've removed it from the latest commit, so i think we should be good!

Comment (luckyj5) [reply glide.yaml]: fyi - there is been a discussion going on in the community regarding case sensitivity of `sirupsen` dep, depends on which one is more efficient, will update it accordingly.

**Comment by luckyj5** [comment]
Thanks for updates @anishp55. Looks good to me.

**Comment by jonathanamartin** [comment]
@luckyj5, @anishp55  Thanks for working on this.  Something we've definitely been needing.  Just pushed an app with the `SPLUNK_INDEX` environment variable but are seeing a null value for the index field.

Anything I'm missing? Any pointers on debugging?

**Comment by anishp55** [comment]
you need ADD_APP_INFO to be set to true.
if that doesn't work let me know and i'll investigate

**Comment by luckyj5** [comment]
Thanks @anishp55. 
@jonathanamartin Yes, ADD_APP_INFO env variable needs to be `true`. 
Also, which version of the nozzle are you using? 
If you are using v 1.1.1, there is a known issue (https://github.com/cloudfoundry-community/splunk-firehose-nozzle/issues/200) which can cause this behavior where the app cache info is not getting populated, hence app data not getting router to the configured index. We are working on fixing the issue right now will roll out another release with the fix soon. 
Thank you!

**Comment by jonathanamartin** [comment]
@luckyj5 we are using v1.1.1.  Verified `ADD_APP_INFO` environment variable is `true`.

**Pull request merged**